internal/telemetry/client.go

```diff
@@ -229,6 +233,16 @@ func (c *client) start(configuration []Configuration, namespace Namespace) {
 	}
 	c.heartbeatInterval = time.Duration(heartbeat) * time.Second
 	c.heartbeatT = time.AfterFunc(c.heartbeatInterval, c.backgroundHeartbeat)
+
+	// app closing should be sent when the app closes gracefully
+	c.sigc = make(chan os.Signal, 1)
+	signal.Notify(c.sigc, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+	go c.appClose()
+}
+
+func (c *client) appClose() {
```

Comment:
we'll probably need to `os.Exit` here to allow the app to close, else it will just hang. 

This means `c.Stop()` needs to block to guarantee `app-closing` is sent. Currently `flush` inside `c.Stop()` triggers another goroutine to actually send the requests. The app may exit before this goroutine is finished.

Also, we might intercept `SIGINT` before the client's own signal handlers and prematurely quit before their cleanup work is done. 

the reverse is also true -- if the client intercepts the specified signals and quits, `app-closing` will never be sent